src/main/java/it/gov/pagopa/pu/pagopapayments/util/RestUtil.java

```diff
@@ -0,0 +1,55 @@
+package it.gov.pagopa.pu.pagopapayments.util;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.HttpServerErrorException;
+import org.springframework.web.client.RestClientException;
+
+import java.util.function.Supplier;
+
+@Slf4j
+public class RestUtil {
+  public static <T> T handleRestException(Supplier<T> in, Supplier<String> logErrorDetail, boolean logElapsed) {
+    long httpCallStart = 0;
+    if(logElapsed)
+      httpCallStart = System.currentTimeMillis();
+    try{
+      return in.get();
+    } catch (HttpServerErrorException he) {
```

Comment:
could be useful to catch NotFound to return null?

Reply:
that's right, it will throw exactly this exception (org.springframework.web.client.HttpClientErrorException$NotFound)